azbfs/url_directory.go

```diff
@@ -174,12 +174,16 @@ func (d DirectoryURL) Rename(ctx context.Context, options RenameDirectoryOptions
 		fileSystemName = &d.filesystem
 	}
 
-	renameSource := "/" + d.filesystem + "/" + d.pathParameter
-
 	urlParts := NewBfsURLParts(d.directoryClient.URL())
 	urlParts.FileSystemName = *fileSystemName
 	urlParts.DirectoryOrFilePath = options.DestinationPath
+	// ensure we use our source's SAS token in the x-ms-rename-source header
+	renameSource := "/" + d.filesystem + "/" + d.pathParameter + "?" + urlParts.SAS.Encode()
 
+	// if we're changing our source SAS to a new SAS in the rename
+	if options.DestinationSas != nil && *options.DestinationSas != "" {
```

Comment:
why does the user need to swap the SAS? Could you please add a comment for context?

Reply:
@zezha-msft I think this is in case the user has separate directory level sas permissions on the destination and the source as opposed to one higher sas permission (container, account, etc) that cover both the destination and source

Reply:
I'll write this in the comment: `In the case the user wants to substitute for example: a dir level sas to an account level sas to rename to the new dir`